### DISCUSSION TITLE
Unable to used Orchard.MarkdownToHtmlAsync in Razor View In Decouple CMS - Orchard Core

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#10621] [Q&A]
Once I reference the module: OrchardCore.MarkDown for the used of Orchard.MarkdownToHtmlAsync in Razor View got that error below is attachment:

![issue](https://user-images.githubusercontent.com/5202291/139756414-0e2a4eee-7b48-4a98-b9d7-33ccd8b53826.PNG)

**Comment by Skrypt** [comment]
I think you need to add a `using OrchardCore.Markdown` in your view or `_viewImport.cshtml` file to use it globally.

**Reply by saleemhaq** [reply]
> using OrchardCore.Markdown

After including still face the same issue:

![issue1](https://user-images.githubusercontent.com/5202291/139785808-cdbef2f5-4f27-4030-8d2a-4cd9c8cbd802.PNG)

**Comment by jtkech** [comment]
I saw the `@page` directive so I assume you defined it under a `/Pages` folder including a dedicated `_ViewImports.cshtml` under the same folder. So, it is not a `Razor.RazorPage` but a `RazorPages.Page` that needs to inherit from.

    @inherits OrchardCore.DisplayManagement.RazorPages.Page

**Reply by saleemhaq** [reply]
> I saw the `@page` directive so I assume you defined it under a `/Pages` folder including a dedicated `_ViewImports.cshtml` under the same folder. So, it is not a `Razor.RazorPage` but a `RazorPages.Page` that needs to inherit from.
> 
> ```
> @inherits OrchardCore.DisplayManagement.RazorPages.Page
> ```

I using the MVC so what should i do?

**Reply by jtkech** [reply]
If you use a `@page`, so under a `/Pages`  folder, just define the following in a view imports under the same folder OR directly in your page `@inherits OrchardCore.DisplayManagement.RazorPages.Page`, not `...Razor.RazorPage<TModel>`.

**Reply by saleemhaq** [reply]
> If you use a `@page`, so under a `/Pages` folder, just define the following in a view imports under the same folder OR directly in your page `@inherits OrchardCore.DisplayManagement.RazorPages.Page`, not `...Razor.RazorPage<TModel>`.

Directly in my page I inject @inherits OrchardCore.DisplayManagement.RazorPages.Page, below is screen shot, please advise further:

![issue-not-calling](https://user-images.githubusercontent.com/5202291/140047522-a3a525d2-fe2c-45d4-b198-b286f1781015.png)

**Reply by ns8482e** [reply]
Guide https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/#loading-a-blog-post-from-a-custom-slug

**Reply by saleemhaq** [reply]
Page need to display inside the parent layout or call the shared layout ? not without the _layout?

**Reply by ns8482e** [reply]
When you inherit from `OrchardCore.DisplayManagement.RazorPages.Page` it will use Theming so it will use "Layout.cshtml" / "Layout.liquid" from your active theme

**Comment by Skrypt** [comment]
Okay, I'm looking at your Solution Explorer screenshot and you seem to have imported Orchard Core modules and abstractions inside your project. Here, you are using a custom solution which may cause you to have all kinds of issues that we can't help you with. First, here, it's not a discussion board for helping you to build your website. We have Gitter for that which makes more sense. Here, it is for discussions about Orchard Core, and not specifically your own website. So, please, we want to help you but you are going in every direction on every answer we give you.

The first thing I would say is, start by sharing your solution on Github so that we can look at it directly and understand what is wrong with it because all the things you are asking for are documented and should work. Thanks for your understanding.
